speechbrain/lobes/models/huggingface_transformers/speecht5.py

```diff
@@ -0,0 +1,460 @@
+"""This lobe enables the integration of huggingface pretrained SpeechT5 models for Automatic Speech Recognition.
+Reference: https://aclanthology.org/2022.acl-long.393.pdf
+
+Transformer from HuggingFace needs to be installed:
+https://huggingface.co/transformers/installation.html
+
+Authors
+ * Haroun Elleuch 2024
+"""
+
+from os import PathLike
+import torch
+import logging
+import gc
+
+from speechbrain.lobes.models.huggingface_transformers.huggingface import (
+    HFTransformersInterface,
+)
+
+from transformers import (
+    SpeechT5ForSpeechToText,
+    SpeechT5Config,
+    GenerationConfig,
+)
+
+from speechbrain.utils.checkpoints import (
+    mark_as_loader,
+    mark_as_saver,
+    register_checkpoint_hooks,
+)
+from speechbrain.utils.fetching import fetch
+
+
+logger = logging.getLogger(__name__)
+
+
+@register_checkpoint_hooks
+class SpeechT5ForASR(HFTransformersInterface):
+    """This lobe enables the integration of HuggingFace and SpeechBrain
+    pretrained SpeechT5 models for Automatic Speech Recognition.
+
+    Source paper SpeechT5: https://aclanthology.org/2022.acl-long.393.pdf
+    Transformer from HuggingFace needs to be installed:
+    https://huggingface.co/transformers/installation.html
+
+    It will download automatically the model from HuggingFace or use a local path.
+
+    Arguments
+    ---------
+    source : str
+        HuggingFace hub name: e.g "microsoft/speecht5_asr"
+    save_path : str
+        save directory of the downloaded model.
+    encoder_only : bool (default: False)
+        If `True`, will only load the (speech) encoder of the SpeechT5 model.
+    freeze : bool (default: True)
+        If `True`, the model is frozen. If `False`, the model will be trained
+        alongside with the rest of the pipeline.
+    freeze_encoder : bool (default: False)
+        If `True`, the model encoder is frozen. If the model is set to encoder only,
+        using the freeze parameter will also freeze the encoder.
+    freeze_feature_extractor: bool (default: False)
+        Whether or not to freeze the feature extractor of SpeechT5.
+    sampling_rate: int (default: 16000)
+        Sampling rate of the audio inputs.
+    output_attentions: bool (default: True)
+        Whether to output the attentions from the encoder.
+    output_all_hiddens: bool (default: True)
+        Whether or not to output all the hidden states from the encoder. They will
+        be stacked if returned.
+
+    Example
+    -------
+    >>> model_hub = "microsoft/speecht5_asr"
+    >>> save_path = "tmp"
+    >>> model = SpeechT5ForASR(model_hub, save_path=save_path)
+    """
+
+    def __init__(
+        self,
+        source: str,
+        save_path: str = "",
+        cache_dir: str = "",
+        freeze: bool = False,
+        freeze_encoder: bool = False,
+        freeze_feature_extractor: bool = False,
+        sampling_rate: int = 16_000,
+        encoder_only: bool = False,
+        output_attentions: bool = True,
```

Comment:
I believe that having output_attentions = True by default will remove the support of flash attention right ?